pages/api/v1/treasury-amount-history/old-treasury-data.ts

```diff
@@ -0,0 +1,214 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+/* eslint-disable sort-keys */
+import storeApiKeyUsage from '~src/api-middlewares/storeApiKeyUsage';
+import withErrorHandling from '~src/api-middlewares/withErrorHandling';
+import { subscanApiHeaders } from '~src/global/apiHeaders';
+import { chainProperties } from '~src/global/networkConstants';
+import messages from '~src/auth/utils/messages';
+import { NextApiRequest, NextApiResponse } from 'next';
+import dayjs from 'dayjs';
+import 'dayjs/locale/en-gb';
+import { firestore_db } from '~src/services/firebaseInit';
+import { IHistoryItem, ITreasuryResponseData } from '~src/types';
+import { isValidNetwork } from '~src/api-utils';
+
+interface IReturnResponse {
+	data?: ITreasuryResponseData[] | null;
+	error?: null | string;
+}
+
+const lastNMonths = 6;
+
+function getMonthRange(monthsAgo: number): { start: string; end: string } {
+	const targetDate = dayjs().subtract(monthsAgo, 'month').set('date', 3);
+	const startDate = targetDate.startOf('day');
+	const endDate = targetDate.endOf('day');
+
+	const start = startDate.format('YYYY-MM-DD');
+	const end = endDate.format('YYYY-MM-DD');
+	return { start, end };
+}
+
+const getMonthName = (date: dayjs.Dayjs): string => {
+	return date.format('MMMM').toLowerCase();
+};
+
+export const aggregateBalances = (data1: ITreasuryResponseData[], data2: ITreasuryResponseData[]): { [key: string]: number } => {
+	const getLatestBalance = (history: IHistoryItem[]): number => {
+		const sortedHistory = history.sort((a, b) => new Date(b.date).getTime() - new Date(a.date).getTime());
+		return parseFloat(sortedHistory[0].balance) || 0;
+	};
+
+	const combinedData: { [key: string]: number } = {};
+
+	const processTreasuryData = (data: ITreasuryResponseData[]) => {
+		data
+			.filter(({ history }) => history && getLatestBalance(history) > 0)
+			.forEach(({ history }) => {
+				history?.forEach(({ date }) => {
+					const key = getMonthName(dayjs(date));
+					const balanceValue = getLatestBalance(history);
+
+					combinedData[key] = (combinedData[key] || 0) + balanceValue;
+				});
+			});
+	};
+
+	processTreasuryData(data1);
+	processTreasuryData(data2);
+
+	return combinedData;
+};
+
+const isDataPresentForCurrentMonth = async (network: string): Promise<boolean> => {
+	const networkRef = firestore_db.collection('networks').doc(network);
+	const networkDoc = await networkRef.get();
+
+	if (networkDoc.exists) {
+		const data = networkDoc.data();
+		const currentMonth = dayjs().format('MMMM').toLowerCase();
+
+		return Object.prototype.hasOwnProperty.call(data?.monthly_treasury_tally, currentMonth);
```

Comment:
avoid using 'hasOwnProperty' this can return true for native methods, please use "in" operator instead